pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/pendingack/impl/PendingAckHandleImpl.java

```diff
@@ -610,6 +570,37 @@ private void addAbortTxnRequest(TxnID txnId, Consumer consumer, long lowWaterMar
         return abortFuture;
     }
 
+    @Override
+    public CompletableFuture<Void> abortTxn(TxnID txnId, Consumer consumer,
+                                                         long lowWaterMark, boolean isInCacheRequest) {
+        CompletableFuture<Void> abortFuture = new CompletableFuture<>();
+        internalPinnedExecutor.execute(() -> {
+            if (!checkIfReady()) {
+                if (state == State.Initializing) {
+                    addAbortTxnRequest(txnId, consumer, lowWaterMark, abortFuture);
+                    return;
+                } else if (state == State.None) {
+                    addAbortTxnRequest(txnId, consumer, lowWaterMark, abortFuture);
+                    initPendingAckStore();
+                    return;
+                } else if (checkIfReady()) {
```

Comment:
we don't need to checkIfReady, because change to ready state use the same thread with abortTxn. like commitTxn or individual ack or cumulative ack.